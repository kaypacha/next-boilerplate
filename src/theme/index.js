import { isString } from 'lodash'

const theme = {
  colors: {
    primary: '#29AC9C',
    secondary: '#CE546D',
    white: 'white',
    black: 'black',
  },
  spacing: (...args) => {
    return args.map(item => (isString(item) ? item : `${8 * item}px`)).join(' ')
  },
  textStyles: {
    h1: {
      fontWeight: 600,
      fontSize: '24px',
      lineHeight: '32px',
    },
    h2: {
      fontWeight: 600,
      fontSize: '18px',
      lineHeight: '24px',
    },
    title: {
      fontWeight: 400,
      fontSize: '16px',
      lineHeight: '24px',
    },
    body1: {
      fontWeight: 400,
      fontSize: '12px',
      lineHeight: '14px',
    },
  },
  textStyle: style => `
    font-size: ${theme.textStyles[style].fontSize};
    font-weight: ${theme.textStyles[style].fontWeight};
    line-height: ${theme.textStyles[style].lineHeight};
    font-family: Open Sans,sans-serif;
  `,
}

export default theme
