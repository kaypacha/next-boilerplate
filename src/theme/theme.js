export const theme = {
  colors: {
    primary: '#29AC9C',
    secondary: '#CE546D',
    white: 'white',
    black: 'black',
  },
  spacing: (...args) => {
    return args.map(item => (typeof item === 'string' ? item : `${8 * item}px`)).join(' ')
  },
  textStyles: {
    h1: {
      fontWeight: 600,
      fontSize: '24px',
      lineHeight: '32px',
    },
    h2: {
      fontWeight: 600,
      fontSize: '18px',
      lineHeight: '24px',
    },
    title: {
      fontWeight: 400,
      fontSize: '16px',
      lineHeight: '24px',
    },
    body1: {
      fontWeight: 400,
      fontSize: '12px',
      lineHeight: '14px',
    },
  },
  textStyle: style => `
    font-size: ${theme.textStyles[style].fontSize};
    font-weight: ${theme.textStyles[style].fontWeight};
    line-height: ${theme.textStyles[style].lineHeight};
    font-family: Open Sans,sans-serif;
  `,
}
