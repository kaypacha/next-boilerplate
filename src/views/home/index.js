import React from 'react'
import { Text } from 'components'
import { DefaultLayout } from 'layouts'

const HomeScreen = () => {
  return (
    <DefaultLayout>
      <Text textStyle="h1" color="white">
        Next is working
      </Text>
    </DefaultLayout>
  )
}

export default HomeScreen
