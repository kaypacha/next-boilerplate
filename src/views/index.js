export { default as HomeScreen } from './home'
