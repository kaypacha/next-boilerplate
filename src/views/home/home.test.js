import { render, screen } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'

import { theme } from 'theme'
import { HomeScreen } from 'views'

const setup = () =>
  render(
    <ThemeProvider theme={theme}>
      <HomeScreen />
    </ThemeProvider>
  )

describe('Text', () => {
  it('renders a text inside Home', () => {
    const wrapper = setup()

    const content = screen.getByText(/Boilerplate by KayPacha/i)

    expect(content).toBeInTheDocument()

    expect(wrapper.container).toMatchSnapshot()
  })
})
