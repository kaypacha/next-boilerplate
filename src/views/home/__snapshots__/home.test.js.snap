// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders a text inside Home 1`] = `
<div>
  <main
    class="sc-8ba3208c-0 eznmQb"
    id="main"
  >
    <span
      class="sc-43f57a72-0 bpNDKf"
      color="white"
    >
      Boilerplate by KayPacha
    </span>
    <span
      class="sc-43f57a72-0 eWlovP"
      color="white"
    >
      Next is working
    </span>
  </main>
</div>
`;
